paderbox/utils/random_utils.py

```diff
@@ -18,80 +20,325 @@ def str_to_random_state(string):
 
 
 def _force_correct_shape(f):
-    """ This decorator sets the seed and fix the snr.
+    """decorator allowing to pass shape as tuple as well
+
+    Args:
+        f: wrapped function
+
+    Returns:
 
-    :param f: Function to be wrapped
-    :return: noise_signal
     """
     @wraps(f)
     def wrapper(*shape, **kwargs):
-        if not shape:
-            shape = (1,)
-        elif isinstance(shape[0], (tuple, list)):
+        if len(shape) > 0 and isinstance(shape[0], (tuple, list)):
+            assert len(shape) == 1, shape
             shape = shape[0]
 
         return f(*shape, **kwargs)
 
     return wrapper
 
 
+def _add_kwarg_dtype(default):
+    """ returns decorator adding the kwarg dtype to the wrapped function and
+    handles the sampling of a certain dtype.
+
+    Args:
+        default: default dtype
+
+    Returns:
+
+    """
+    def _decorator(f):
+        @wraps(f)
+        def wrapper(*shape, dtype=default, **kwargs):
+
+            def _f(dtype_local):
+                return np.array(f(*shape, **kwargs), dtype=dtype_local)
+
+            if dtype in (np.float32, np.float64):
+                return _f(dtype)
+            elif dtype is np.complex64:
+                return _f(np.float32) + 1j * _f(np.float32)
+            elif dtype is np.complex128:
+                return _f(np.float64) + 1j * _f(np.float64)
+            else:
+                raise ValueError(f'Invalid dtype {dtype}')
+
+        return wrapper
+    return _decorator
+
+
 @_force_correct_shape
-def uniform(*shape, data_type=np.complex128):
+@_add_kwarg_dtype(default=np.float64)
+def uniform(*shape, low=-1., high=1.):
+    """
 
-    def _uniform(data_type_local):
-        return np.random.uniform(-1, 1, shape).astype(data_type_local)
+    Args:
+        *shape:
+        low:
+        high:
+        dtype:
 
-    if data_type in (np.float32, np.float64):
-        return _uniform(data_type)
-    elif data_type is np.complex64:
-        return _uniform(np.float32) + 1j * _uniform(np.float32)
-    elif data_type is np.complex128:
-        return _uniform(np.float64) + 1j * _uniform(np.float64)
+    Returns:
+
+    >>> x = uniform()
+    >>> x.ndim
+    0
+    >>> x.dtype
+    dtype('float64')
+    >>> x = uniform(2, 3)
+    >>> x.shape, x.dtype
+    ((2, 3), dtype('float64'))
+    >>> x = uniform(2, 3, dtype=np.complex128)
+    >>> x.shape, x.dtype
+    ((2, 3), dtype('complex128'))
+    """
+    return np.random.uniform(low=low, high=high, size=None if len(shape) == 0 else shape)
```

Comment:
I thought `size=shape` should work when `len(shape) == 0`.

Reply:
yes you are right